Foundation/NSMeasurement.swift

```diff
@@ -23,21 +23,93 @@ open class NSMeasurement : NSObject, NSCopying, NSSecureCoding {
+    open func adding(_ rhs: Measurement<Unit>) -> Measurement<Unit> {
+        if self.unit.isEqual(rhs.unit) {
+            return Measurement(value: self.doubleValue + rhs.value, unit: self.unit)
+        } else {
+            guard let dimension = unit as? Dimension,
+                    let otherDimension = rhs.unit as? Dimension else {
+                        fatalError("Cannot convert differing units that are non-dimensional! lhs: \(type(of: unit)) rhs: \(type(of: rhs.unit))")
+                }
+            let selfValueInTermsOfBase = dimension.converter.baseUnitValue(fromValue: self.doubleValue)
+            let rhsValueInTermsOfBase = otherDimension.converter.baseUnitValue(fromValue: rhs.value)
+            return Measurement(value: selfValueInTermsOfBase + rhsValueInTermsOfBase, unit: type(of: dimension).baseUnit())
+        }
+    }
     
-    open func subtracting(_ measurement: Measurement<Unit>) -> Measurement<Unit> { NSUnimplemented() }
+    open func subtracting(_ rhs: Measurement<Unit>) -> Measurement<Unit> {
+        if self.unit.isEqual(rhs.unit) {
+            return Measurement(value: self.doubleValue - rhs.value, unit: self.unit)
+        } else {
+            guard let dimension = unit as? Dimension,
+                    let otherDimension = rhs.unit as? Dimension else {
+                        fatalError("Cannot convert differing units that are non-dimensional! lhs: \(type(of: unit)) rhs: \(type(of: rhs.unit))")
+                }
+            let selfValueInTermsOfBase = dimension.converter.baseUnitValue(fromValue: self.doubleValue)
+            let rhsValueInTermsOfBase = otherDimension.converter.baseUnitValue(fromValue: rhs.value)
+            return Measurement(value: selfValueInTermsOfBase - rhsValueInTermsOfBase, unit: type(of: dimension).baseUnit())
+        }
+    }
     
-    open func copy(with zone: NSZone? = nil) -> Any { NSUnimplemented() }
+    open func copy(with zone: NSZone? = nil) -> Any { return self }
     
     open class var supportsSecureCoding: Bool { return true }
     
-    open func encode(with aCoder: NSCoder) { NSUnimplemented() }
+    open func encode(with aCoder: NSCoder) {
+        guard aCoder.allowsKeyedCoding else {
+            preconditionFailure("Unkeyed coding is unsupported.")
+        }
+        aCoder.encode(self.doubleValue, forKey:"NS.value")
+        aCoder.encode(self.unit, forKey:"NS.unit")
+    }
     
-    public required init?(coder aDecoder: NSCoder) { NSUnimplemented() }
+    public required init?(coder aDecoder: NSCoder) {
+        guard aDecoder.allowsKeyedCoding else {
+            preconditionFailure("Unkeyed coding is unsupported.")
+        }
+        let doubleValue = aDecoder.decodeDouble(forKey: "NS.value")
+        let possibleUnit = aDecoder.decodeObject(forKey: "NS.unit")
+        guard let unit = possibleUnit as? Unit else {
+            return nil // or should we `fatalError()`?
```

Comment:
What does Darwin Foundation do?

Reply:
One thing it does is validation, e.g. @"Cannot add measurements of differing unit types!"